### PR TITLE
Add faucet plugin

### PR DESCRIPTION
**Pull request by muXxer**
This PR adds the new faucet plugin.

Free coins for everyone :trollface:

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -210,6 +210,16 @@
     "workers": 0,
     "autostart": false
   },
+  "faucet": {
+    "hrpNetworkPrefix": "atoi",
```
Comment (alexsporn) [inline config.json]: this is wrong for config.json

File: plugins/faucet/params.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	CfgFaucetBatchTimeout = "faucet.batchTimeout"
+	// the amount of workers used for calculating PoW when issuing faucet messages.
+	CfgFaucetPoWWorkerCount = "faucet.powWorkerCount"
+)
+
+var params = &node.PluginParams{
+	Params: map[string]*flag.FlagSet{
+		"nodeConfig": func() *flag.FlagSet {
+			fs := flag.NewFlagSet("", flag.ContinueOnError)
+			fs.Int64(CfgFaucetAmount, 10000000, "the amount of funds the requester receives if the target address has less funds than the given amount")
```
Comment (alexsporn) [inline plugins/faucet/params.go]: `has less funds than the given amount` -> -`has less funds than the maximum allowed`

File: plugins/faucet/params.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const (
+	// the amount of funds the requester receives if the target address has less funds than the given amount.
+	CfgFaucetAmount = "faucet.amount"
+	// the amount of funds the requester receives if the target address has more funds than the given amount.
+	CfgFaucetSmallAmount = "faucet.smallAmount"
+	// the maximum allowed amount of funds on the target address.
+	CfgFaucetMaxAddressBalance = "faucet.maxAddressBalance"
+	// the maximum output count per faucet message.
+	CfgFaucetMaxOutputCount = "faucet.maxOutputCount"
```
Comment (alexsporn) [inline plugins/faucet/params.go:20]: why is this configurable? to avoid big pow?

Comment (muXxer) [reply plugins/faucet/params.go:20]: I can't remember but I just say yes :trollface:

File: plugins/faucet/params.go (hunk 2 above)
Comment (muXxer) [reply plugins/faucet/params.go]: We have two thresholds, for example the "amount" is 10 Mi.

If someone has less than 10 Mi, he gets 10 Mi on his address.
If he has more than 10 Mi already, he only gets the small amount (1 Mi) until the max is reached (20 Mi).

Comment (alexsporn) [reply plugins/faucet/params.go]: I know, but your sentence in the help does not make sense. 
what is "the given amount" in that sentence?

**Review by alexsporn — APPROVED** [review]

**Pull request merged**